Assets/Content/Items/Functional/Tools/TableConstructer.cs

```diff
@@ -31,10 +34,26 @@ public void Interact()
 
             var tile = targetTile.Tile;
 
-            if (tile.fixture == tableToConstruct) // Deconstruct
-                tile.fixture = null;
-            else // Construct
-                tile.fixture = tableToConstruct;
+            // The player using this item.
+            var player = transform.root;
+
+            // Cancel interaction if the target tile is outside the build range.
+            if (Vector3.Distance(player.transform.position, targetTile.transform.position) > buildDistance)
+            {
+                return;
+            }
+
+            if (tile.fixture != null) // If there is a fixture on the place
+            {
+                if (tile.fixture == tableToConstruct) // If the fixture is a table
+                {
+                    tile.fixture = null; // Deconstruct
+                }
+            }
+            else // If there is no fixture on place
+            {
+                tile.fixture = tableToConstruct; // Construct
+            }
```

Comment:
Because now nothing happens when `tile.fixture != null && tile.fixture != tableToConstruct`, this case should be added to CanInteract

Reply:
Code was moved to proper place in the latest commit.